### PR TITLE
Add initial distribution parameter types

### PR DESCRIPTION
**Pull request by canyon289**
Per comment https://github.com/pymc-devs/pymc/pull/5298#discussion_r778537160

Let me know if I got it right @ricardoV94

### CODE REVIEW

File: pymc/distributions/continuous.py [hunk 1]
```diff
@@ -594,7 +597,7 @@ class TruncatedNormalRV(RandomVariable):
     def rng_fn(
         cls,
         rng: np.random.RandomState,
-        mu: Union[np.ndarray, float],
+        mu: DIST_PARAMETER_TYPES,
```
Comment (ricardoV94) [inline pymc/distributions/continuous.py]: In the `rng_fn` the previous type hints were correct, it's in the `.dist` classmethod in the distribution below that the `TensorVariable` also shows up as a possibility

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Just to not be mistakenly merged in the meantime

File: pymc/distributions/continuous.py [hunk 2]
```diff
@@ -692,12 +692,12 @@ class TruncatedNormal(BoundedContinuous):
     @classmethod
     def dist(
         cls,
-        mu: Optional[Union[float, np.ndarray]] = None,
-        sigma: Optional[Union[float, np.ndarray]] = None,
-        tau: Optional[Union[float, np.ndarray]] = None,
-        sd: Optional[Union[float, np.ndarray]] = None,
-        lower: Optional[Union[float, np.ndarray]] = None,
-        upper: Optional[Union[float, np.ndarray]] = None,
+        mu: Optional[DIST_PARAMETER_TYPES] = None,
```
Comment (canyon289) [inline pymc/distributions/continuous.py:695]: Thanks for previous comment @ricardoV94. Hows it looking now?

One question, are dist parameter always optional? If so I can move the optional flag to the definition

Comment (ricardoV94) [reply pymc/distributions/continuous.py:695]: No they are not always optional. Most times they are required actually

Comment (canyon289) [reply pymc/distributions/continuous.py:695]: Hows it look now?

Comment (ricardoV94) [reply pymc/distributions/continuous.py:695]: Good!

Comment (canyon289) [reply pymc/distributions/continuous.py:695]: 🎉  can we merge? Ill create an issue ticket for data umbrella sprint too have others add to the other parameters.

Thanks Ricardo

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**